connect/api/src/main/java/org/apache/kafka/connect/rest/extension/entities/ConnectorStateDetail.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.rest.extension.entities;
+
+
+import java.util.List;
+import java.util.Objects;
+
+public class ConnectorStateDetail {
```

Comment:
Why not an interface? That would offer us so much more flexibility in the implementation, and all of the implementation details can be hidden from the public API. By having the nested classes here, they are in the public API and need to be managed through KIPs.

Reply:
The nested classes are also technically part of the public API since user can access it. But IMO, having interfaces for POJO or entities might be a little bit of overhead.  The only thing we can hide by making it an interface is the constructor.

Reply:
Using an interface gives us more options on the implementation side, and it helps clarify intensions more explicitly and minimally, without exposing implementation details.